### ISSUE TITLE
Can't write to 777 files owned by other users in 3.6.0 with --writable-tmpfs (worked in 3.5.3)

### ISSUE THREAD

**Issue by adiemu5** [apptainer/singularity#5455]
### Version of Singularity:

3.6.0

### Expected behavior

Using this definition:
```
BootStrap: docker
From: alpine:latest

%post
        mkdir /perm_test
        touch /perm_test/root.log
        chown root:root /perm_test/root.log
        chmod 777 /perm_test/root.log

        touch /perm_test/nobody.log
        chown nobody:nobody /perm_test/nobody.log
        chmod 777 /perm_test/nobody.log
```
with 3.5.3, I get:
```
$ id
uid=10000(test) gid=10000(test) groups=10000(test) context=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023
$ singularity --version
singularity version 3.5.3-1.1.el7
$ singularity shell --writable-tmpfs test.sif
Singularity> ls -l /perm_test/
total 1
-rwxrwxrwx    1 nobody   nobody           0 Jul 24 14:29 nobody.log
-rwxrwxrwx    1 root     root             0 Jul 24 14:29 root.log
Singularity> touch /perm_test/*
Singularity> 
```

### Actual behavior

In 3.6.0, touching (or trying to write to) these files fails with a permission error:

```
$ id
uid=10000(test) gid=10000(test) groups=10000(test) context=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023
$ singularity --version
singularity version 3.6.0-1.el7
$ singularity shell --writable-tmpfs test.sif
Singularity> ls -l /perm_test/
total 1
-rwxrwxrwx    1 nobody   nobody           0 Jul 24 14:27 nobody.log
-rwxrwxrwx    1 root     root             0 Jul 24 14:27 root.log
Singularity> touch /perm_test/*
touch: /perm_test/nobody.log: Permission denied
touch: /perm_test/root.log: Permission denied
Singularity> 
```

### Steps to reproduce this behavior

Use the definition file and commands provided above.

### What OS/distro are you running

Distro: CentOS 7.8
Kernel: 3.10.0-1127.13.1.el7.x86_64

### How did you install Singularity

EPEL 7 RPM - singularity-3.6.0-1.el7.x86_64

**Comment by adiemu5** [comment]
For what it's worth, I was able to bisect the source of this regression to commit fdd5917.

Interestingly, with commit fdd5917 the example above starts returning "Read-only filesystem" as the error.  It is actually only in commit 1349a3473bff9b11375b00021ca018b81aa5c90c that the error finally shifts to "Permission denied" as seen in 3.6.0.

In terms of real-world impact, this prevents us from being able to deploy Singularity 3.6.x on our HPC cluster as it breaks existing workflows for containers like pgAdmin and Postgres which have existing files owned by root or postgres that need to be (non-persistently) writable by the container user.  (Which we handled in 3.5.x via `chmod 777 file` in %post)

**Comment by dtrudg** [comment]
@cclerget - do you have any thoughts on this? I'm able to replicate.

**Comment by cclerget** [comment]
@adiemu5 @dctrud Should be fixed by #5463

**Comment by adiemu5** [comment]
Confirmed this is fixed by #5463, assuming I tested correctly.  (The build ends up listing its version as `3.5.2+661-g5417e6b` which seems odd, as I'd have expected `3.6.1+something`)
